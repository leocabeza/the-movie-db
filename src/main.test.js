import entryPoint from './main';

describe('main api', () => {
  it('returns a function', () => {
    expect(typeof entryPoint.v3).toEqual('function');
    expect(typeof entryPoint.v4).toEqual('function');
  });

  it('returns an object with methods', () => {
    expect(entryPoint.v3('v3 api key')).toMatchSnapshot();
    expect(entryPoint.v4('v4 api key')).toMatchSnapshot();
  });
});
