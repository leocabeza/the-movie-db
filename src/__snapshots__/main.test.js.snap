// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main api returns an object with methods 1`] = `
Object {
  "account": Object {
    "addToWatchlist": [Function],
    "details": [Function],
    "favoriteMovies": [Function],
    "favoriteTvShows": [Function],
    "lists": [Function],
    "markAsFavorite": [Function],
    "movieWatchlist": [Function],
    "ratedMovies": [Function],
    "ratedTvEpisodes": [Function],
    "ratedTvShows": [Function],
    "tvShowWatchlist": [Function],
  },
  "authentication": Object {
    "logout": [Function],
    "newGuestSession": [Function],
    "newSession": [Function],
    "newToken": [Function],
    "sessionConvert": [Function],
    "validateWithLogin": [Function],
  },
  "certification": Object {
    "movies": [Function],
    "tvShows": [Function],
  },
  "change": Object {
    "movies": [Function],
    "people": [Function],
    "tvShows": [Function],
  },
  "collection": Object {
    "details": [Function],
    "images": [Function],
    "translations": [Function],
  },
  "company": Object {
    "alternativeNames": [Function],
    "details": [Function],
    "images": [Function],
  },
  "configuration": Object {
    "api": [Function],
    "countries": [Function],
    "jobs": [Function],
    "languages": [Function],
    "primaryTranslations": [Function],
    "timezones": [Function],
  },
  "credit": Object {
    "details": [Function],
  },
  "discover": Object {
    "movie": [Function],
    "tvShows": [Function],
  },
  "find": Object {
    "byId": [Function],
  },
  "genre": Object {
    "movieList": [Function],
    "tvList": [Function],
  },
  "guestSession": Object {
    "ratedMovies": [Function],
    "ratedTvEpisodes": [Function],
    "ratedTvShows": [Function],
  },
  "keyword": Object {
    "details": [Function],
    "movies": [Function],
  },
  "list": Object {
    "addMovie": [Function],
    "clear": [Function],
    "create": [Function],
    "details": [Function],
    "itemStatus": [Function],
    "remove": [Function],
    "removeMovie": [Function],
  },
  "movie": Object {
    "accountStates": [Function],
    "alternativeTitles": [Function],
    "changes": [Function],
    "credits": [Function],
    "deleteRating": [Function],
    "details": [Function],
    "externalIds": [Function],
    "images": [Function],
    "keywords": [Function],
    "latest": [Function],
    "lists": [Function],
    "nowPlaying": [Function],
    "popular": [Function],
    "rate": [Function],
    "recommendations": [Function],
    "releaseDates": [Function],
    "reviews": [Function],
    "similar": [Function],
    "topRated": [Function],
    "translations": [Function],
    "upcoming": [Function],
    "videos": [Function],
  },
  "network": Object {
    "alternativeNames": [Function],
    "details": [Function],
    "images": [Function],
  },
  "people": Object {
    "changes": [Function],
    "combinedCredits": [Function],
    "details": [Function],
    "externalIds": [Function],
    "images": [Function],
    "latest": [Function],
    "movieCredits": [Function],
    "popular": [Function],
    "taggedImages": [Function],
    "translations": [Function],
    "tvCredits": [Function],
  },
  "review": Object {
    "details": [Function],
  },
  "search": Object {
    "collections": [Function],
    "companies": [Function],
    "keywords": [Function],
    "movies": [Function],
    "multi": [Function],
    "people": [Function],
    "tv": [Function],
  },
  "trending": Object {
    "items": [Function],
  },
  "tv": Object {
    "accountStates": [Function],
    "airingToday": [Function],
    "alternativeTitles": [Function],
    "changes": [Function],
    "contentRatings": [Function],
    "credits": [Function],
    "deleteRating": [Function],
    "details": [Function],
    "episodeGroups": [Function],
    "externalIds": [Function],
    "images": [Function],
    "keywords": [Function],
    "latest": [Function],
    "onTheAir": [Function],
    "popular": [Function],
    "rate": [Function],
    "recommendations": [Function],
    "reviews": [Function],
    "screenedTheatrically": [Function],
    "similar": [Function],
    "topRated": [Function],
    "translations": [Function],
    "videos": [Function],
  },
  "tvEpisode": Object {
    "accountStates": [Function],
    "changes": [Function],
    "credits": [Function],
    "deleteRating": [Function],
    "details": [Function],
    "externalIds": [Function],
    "images": [Function],
    "rate": [Function],
    "translations": [Function],
    "videos": [Function],
  },
  "tvEpisodeGroup": Object {
    "details": [Function],
  },
  "tvSeason": Object {
    "accountStates": [Function],
    "changes": [Function],
    "credits": [Function],
    "details": [Function],
    "externalIds": [Function],
    "images": [Function],
    "videos": [Function],
  },
}
`;

exports[`main api returns an object with methods 2`] = `
Object {
  "account": Object {
    "favoriteMovies": [Function],
    "favoriteTvShows": [Function],
    "lists": [Function],
    "movieRecommendations": [Function],
    "movieWatchlist": [Function],
    "ratedMovies": [Function],
    "ratedTvShows": [Function],
    "tvShowRecommendations": [Function],
    "tvShowWatchlist": [Function],
  },
  "authentication": Object {
    "createAccessToken": [Function],
    "createRequestToken": [Function],
    "deleteAccessToken": [Function],
  },
  "list": Object {
    "addItems": [Function],
    "clearItems": [Function],
    "create": [Function],
    "deleteItems": [Function],
    "details": [Function],
    "itemStatus": [Function],
    "remove": [Function],
    "update": [Function],
    "updateItems": [Function],
  },
}
`;
